app/components/common/Navigation.vue

```diff
@@ -0,0 +1,534 @@
+   */
+  export default Vue.extend({
+    computed: {
+      isOldBrowser () {
+        if (window.features.china && $.browser) {
+          if (!($.browser.webkit || $.browser.mozilla || $.browser.msedge)) {
+            return true
+          }
+          const majorVersion = $.browser.versionNumber
+          if ($.browser.mozilla && majorVersion < 25) {
+            return true
+          }
+          if ($.browser.chrome && majorVersion < 72) { // forbid some chinese browser
+            return true
+          }
+          if ($.browser.safari && majorVersion < 6) { // 6 might have problems with Aether, or maybe just old minors of 6: https://errorception.com/projects/51a79585ee207206390002a2/errors/547a202e1ead63ba4e4ac9fd
+            return true
+          }
+        }
+        return false
+      },
+
+      isCodeCombat () {
+        if (LOCALHOST_FAKE_HOST() === CODECOMBAT) {
+          return true
+        }
+        return document.location.host.includes(CODECOMBAT) || document.location.host.includes(CODECOMBAT_CHINA)
+      },
+
+      isOzaria () {
+        if (LOCALHOST_FAKE_HOST() === OZARIA) {
+          return true
+        }
+        return document.location.host.includes(OZARIA) || document.location.host.includes(OZARIA_CHINA)
+      },
+
+      cocoBaseURL () {
+        if (this.isCodeCombat) {
+          return ''
+        }
+
+        if (!application.isProduction()) {
+          return `${document.location.protocol}//codecombat.com`
+        }
+
+        // We are on ozaria domain.
+        return `${document.location.protocol}//${document.location.host}`
+          .replace(OZARIA, CODECOMBAT)
+          .replace(OZARIA_CHINA, CODECOMBAT_CHINA)
+      },
+
+      ozBaseURL () {
+        if (this.isOzaria) {
+          return ''
+        }
+
+        if (!application.isProduction()) {
+          return `${document.location.protocol}//ozaria.com`
+        }
+
+        // We are on codecombat domain.
+        return `${document.location.protocol}//${document.location.host}`
+          .replace(CODECOMBAT, OZARIA)
+          .replace(CODECOMBAT_CHINA, OZARIA_CHINA)
+      }
+    },
+
+    created () {
+      // Bind the global values to the vue component.
+      this.me = me
+      this.document = window.document
+      this.serverConfig = window.serverConfig
+      this.serverSession = window.serverSession
+      this.CODECOMBAT = CODECOMBAT
+      this.OZARIA = OZARIA
+    },
+
+    methods: {
+      /**
+       * This is used to highlight nav routes we are currently on.
+       * It can optionally also check if the user is on codecombat or ozaria.
+       */
+      checkLocation (route, host = undefined) {
+        let hostCheck = true
+        if (host === CODECOMBAT) {
+          hostCheck = this.isCodeCombat
+        } else if (host === OZARIA) {
+          hostCheck = this.isOzaria
+        }
+        return hostCheck && document.location.href.search(route) >= 0
+      },
+
+      isAnonymous () {
```

Comment:
Why we do we put one accessor method here for `isAnonymous()`  but leave `me.isStudent()` and other `me` methods directly inline? Do we need this method?

Reply:
Great catch. I initially was going to migrate all of the methods away but I think it adds unnecessary noise. Will delete this and use `me.isAnonymous()` instead.